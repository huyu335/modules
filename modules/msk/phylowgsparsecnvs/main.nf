process PHYLOWGSPARSECNVS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'mskcc/phylowgs:v1.4-msk':
        'mskcc/phylowgs:v1.4-msk' }"

    input:
    tuple val(meta), path(facetsgenelevel)

    output:
    tuple val(meta), path("cnvs.txt"), emit: cnv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    python2 \\
        /usr/bin/parser/parse_cnvs.py \\
        ${args} \\
        ${facetsgenelevel}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgsparsecnvs: \$PHYLOWGS_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch cnvs.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgsparsecnvs: \$PHYLOWGS_TAG
    END_VERSIONS
    """
}
