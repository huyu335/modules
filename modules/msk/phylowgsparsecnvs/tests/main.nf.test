nextflow_process {

    name "Test Process PHYLOWGSPARSECNVS"
    script "../main.nf"
    process "PHYLOWGSPARSECNVS"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "phylowgsparsecnvs"

    test("phylowgsparsecnvs - gene_level") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['facets_gene_level_txt'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

    test("phylowgsparsecnvs - gene_level") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['facets_gene_level.txt'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
