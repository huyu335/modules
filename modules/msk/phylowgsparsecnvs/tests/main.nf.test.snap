{
    "phylowgsparsecnvs - txt": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "cnv": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-06T13:25:07.411262"
    }
}